### ISSUE TITLE
NTML Auth Type

### ISSUE THREAD

**Issue by anonymouz4** [rofl0r/proxychains-ng#240]
Is there a way to authenticate via NTML Hash of user/pass just like when using http proxy with openvpn?

**Comment by rofl0r** [comment]
i don't think so. is that an extended auth feature (i.e. not "basic" auth) of HTTP ?

**Comment by anonymouz4** [comment]
Well I'm not an expert, but I guess the username is sent in clear text and pass is just hashed.
Since OpenVPN is opensource, wouldn't it be possible to just copy the authentication code?
Because in ovpn config you can choose: basic, ntml and auto(detect)

**Comment by anonymouz4** [comment]
Well I'm not an expert, but I guess the username is sent in clear text and pass is just hashed.
Since OpenVPN is opensource, wouldn't it be possible to just copy the authentication code?
Because in ovpn config you can choose: basic, ntml and auto(detect)

This is really important bc as far as I know, every Windows based Proxyserver uses ntml auth

**Comment by rofl0r** [comment]
> username is sent in clear text and pass is just hashed

that's equally safe as sending the password in plaintext, since an adversary can capture the hash and send it again, unless the hash involves a cryptographic nonce sent from the server... e.g. server sends some random bytes and then you concatenate those bytes with the password and take the hash from the entire thing

**Comment by rofl0r** [comment]
so what is that proxyserver? http ? socks5 ? what's its name so we can do some research ?

**Comment by anonymouz4** [comment]
HTTP

**Comment by anonymouz4** [comment]
HTTP
Port 3128
No need to do research, jut look at openvpn code

**Comment by rofl0r** [comment]
1. the thing is called NTLM not NTML
2. since i don't have a server to test, i can't implement this feature
3. protocol is described here https://msdn.microsoft.com/en-us/library/dd925287(v=office.12).aspx
4. if someone wants to give it a go, PRs are welcome

**Comment by anonymouz4** [comment]
1. sorry for the typo
2. my only guess was to see if code can be copied from [here](https://github.com/OpenVPN/openvpn/search?q=ntlm&unscoped_q=ntlm), but if that isn't possible then hopefully smo. who has experience can give it a go
